src/lib.rs

```diff
@@ -387,6 +388,17 @@ impl Server {
             }
         }
     }
+
+    /// Same as `recv()` but doesn't block longer than timeout
+    pub fn try_recv_timeout(&self, timeout: Duration) -> IoResult<Option<Request>> {
+        loop {
```

Comment:
I think that's a mistake in the existing code. The other functions have this unnecessary loop as well, and this one was probably copy-pasted from the others.